scripts/zones/Bostaunieux_Oubliette/npcs/Novalmauge.lua

```diff
@@ -15,39 +15,38 @@ require("scripts/globals/quests")
 
 local path =
 {
-    41.169430, -24.000000, 19.860674,
-    42.256676, -24.000000, 19.885197,
-    41.168694, -24.000000, 19.904638,
-    21.859211, -24.010996, 19.792259,
-    51.917370, -23.924366, 19.970068,
-    74.570229, -24.024828, 20.103880,
-    44.533886, -23.947662, 19.926519
+    [1] = {20.000, -24.000, 20.000, 0},
+    [2] = {50.000, -24.000, 20.000, 0},
+    [3] = {75.000, -24.000, 20.000, 0},
+    [2] = {50.000, -24.000, 20.000, 0}
 }
 
 local wsQuest = tpz.wsquest.spiral_hell
 
 function onSpawn(npc)
     npc:initNpcAi()
-    npc:setPos(tpz.path.first(path))
+    npc:setPos(path[1][1], path[1][2], path[1][3], 0)
+    npc:speed(12)
+    npc:setPathPoint(1)
```

Comment:
Thoughts of baking this into `initNpcAi`? Can make it take a table, grab the first point while in core, set the pos, and also potentially take a walking speed as well to set as a local var.

Reply:
i did change the name of initNpcAI to initNpcPathing and you can now call it as npc:initNpcPathing() or pass an x,y,z to set the start position so npc:initNpcPathing(x,y,z) and also it will set the path point to 1 as default, but you can pass a 4th param to set the point also so npc:initNpcPathing(x, y, z, 5) will set the position and set the starting path point to 5.